src/KintoID.sol

```diff
@@ -180,6 +180,25 @@ contract KintoID is
 
     /* ============ Burn ============ */
 
+    /**
+     * @dev Transfers the NFT from the old account to the new account
+     * @param _from Old address
+     * @param _to New address
+     */
+    function transferOnRecovery(address _from, address _to) external override {
+        require(balanceOf(_from) > 0 && balanceOf(_to) == 0, "Invalid transfer");
+        require(
+            msg.sender == walletFactory || hasRole(DEFAULT_ADMIN_ROLE, msg.sender),
```

Comment:
alternatively, we could avoid adding the `walletFactory` here and just grating the walletFactory the `ADMIN_ROLE`

Reply:
better to not give the factory all the other powers just in case the factory gets compromised

Reply:
sg!